blocks/generator.go

```diff
@@ -131,13 +136,44 @@ func (g *Generator) calculateSmesherRewards(logger log.Log, layerID types.LayerI
 			logger.With().Warning("proposal ATX not found", p.ID(), p.AtxID, log.Err(err))
 			return nil, fmt.Errorf("block gen get ATX: %w", err)
 		}
+		ballot := &p.Ballot
+		weightPer, err := proposals.ComputeWeightPerEligibility(g.atxDB, g.meshDB, ballot, g.cfg.LayerSize, g.cfg.LayersPerEpoch)
+		if err != nil {
+			logger.With().Error("failed to calculate weight per eligibility", p.ID(), log.Err(err))
+			return nil, err
+		}
+		logger.With().Debug("weight per eligibility", p.ID(), log.Stringer("weight_per", weightPer))
+		actual := weightPer.Mul(util.WeightFromUint64(uint64(len(ballot.EligibilityProofs))))
+		if _, ok := weights[atx.Coinbase]; !ok {
+			weights[atx.Coinbase] = actual
+			coinbases = append(coinbases, atx.Coinbase)
+		} else {
+			weights[atx.Coinbase].Add(actual)
+		}
+		events.ReportProposal(events.ProposalIncluded, p)
+	}
+	// make sure we output coinbase in a stable order.
+	sort.Slice(coinbases, func(i, j int) bool {
+		return bytes.Compare(coinbases[i].Bytes(), coinbases[j].Bytes()) < 0
+	})
+	rewards := make([]types.AnyReward, 0, len(weights))
+	for _, coinbase := range coinbases {
+		weight, ok := weights[coinbase]
+		if !ok {
+			g.logger.With().Fatal("coinbase missing", coinbase)
+		}
+		wb, err := weight.GobEncode()
```

Comment:
this is definitely wrong. it is weird that we use []byte to store weight, but why it gets encoded with gob